### PR TITLE
perf: Remove parsing of integer literals

### PR DESCRIPTION
**Pull request by big-andy-coates**
# Description 

While reviewing another PR I noticed that the code in the `SqlToJavaVisitor` converted INT and BIGINT literals to strings and then built java code that would parse these strings when processing each row of data. This adds unnecessary overhead to the row processing and so has been replaced with just the int/long value.

Also improved error handling of numeric parsing: DOUBLEs now throw a parse exception if they result in NaN or Infinity and parse exceptions include correct location.

### Testing done 

Added QTT test

`mvn test`

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-functional-tests/src/test/resources/query-validation-tests/literals.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{
+  "tests": [
```
Comment (rodesai) [inline ksql-functional-tests/src/test/resources/query-validation-tests/literals.json]: I think we should also add a case for a 0-prefixed literal (e.g. 0123 should be treated as 123)

File: ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java [hunk 2]
```diff
@@ -1044,7 +1044,20 @@ public Node visitIntegerLiteral(final SqlBaseParser.IntegerLiteralContext contex
 
     @Override
     public Node visitDecimalLiteral(final SqlBaseParser.DecimalLiteralContext context) {
-      return new DoubleLiteral(getLocation(context), context.getText());
+      final Optional<NodeLocation> location = getLocation(context);
+
+      try {
+        final double value = Double.parseDouble(context.getText());
+        if (Double.isNaN(value)) {
```
Comment (rodesai) [inline ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java]: add unit tests for these cases?

Comment (big-andy-coates) [reply ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java]: done.

**Comment by big-andy-coates** [comment]
Why no green tick @rodesai ?  I'm going to assume you meant to give one...

**Pull request merged**